src/test/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidatorTest.java

```diff
@@ -29,20 +34,113 @@
 
+        validator.validateTotalCurrentAssets(currentPeriod, errors);
+
+        assertTrue(errors.containsError(
+                new Error("incorrect_total", CURRENT_ASSETS_TOTAL_PATH,
+                        LocationType.JSON_PATH.getValue(),
+                        ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Test current assets validation with empty values")
+    public void validateTotalCurrentAssetsWithEmptyValues() {
+
+        CurrentAssets currentAssets = new CurrentAssets();
+        currentAssets.setStocks(null);
+        currentAssets.setDebtors(null);
+        currentAssets.setCashAtBankAndInHand(5L);
+        currentAssets.setTotalCurrentAssets(10L);
+
+        balanceSheet.setCurrentAssets(currentAssets);
+        currentPeriod.setBalanceSheet(balanceSheet);
+        ReflectionTestUtils.setField(validator, "incorrectTotal", "incorrect_total");
 
-       Errors errors =  validator.validateCurrentPeriod(currentPeriod);
+        validator.validateTotalCurrentAssets(currentPeriod, errors);
 
         assertTrue(errors.containsError(
-            new Error("incorrect_total", TOTAL_PATH,
-                LocationType.JSON_PATH.getValue(),
-                ErrorType.VALIDATION.getType())));
+                new Error("incorrect_total", CURRENT_ASSETS_TOTAL_PATH,
+                        LocationType.JSON_PATH.getValue(),
+                        ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Test total current assets no error with correct values")
+    public void validateTotalCurrentAssetsWithCorrectValues() {
+
+        CurrentAssets currentAssets = new CurrentAssets();
+        currentAssets.setStocks(5L);
+        currentAssets.setDebtors(5L);
+        currentAssets.setCashAtBankAndInHand(5L);
+        currentAssets.setTotalCurrentAssets(15L);
+
+        balanceSheet.setCurrentAssets(currentAssets);
+
+        currentPeriod.setBalanceSheet(balanceSheet);
+        ReflectionTestUtils.setField(validator, "incorrectTotal", "incorrect_total");
+
+        validator.validateTotalCurrentAssets(currentPeriod, errors);
+
+        assertFalse(errors.hasErrors());
+
+    }
+
+    @Test
+    @DisplayName("Test validate whole current period")
+    public void validateCurrentPeriod(){
+
+        addInvalidFixedAssetsToBalanceSheet();
+        addInvalidCurrentAssetsToBalanceSheet();
+
+        currentPeriod.setBalanceSheet(balanceSheet);
+
+        ReflectionTestUtils.setField(validator, "incorrectTotal", "incorrect_total");
+
+        errors = validator.validateCurrentPeriod(currentPeriod);
+
+        assertTrue(errors.hasErrors());
+        assertEquals(2, errors.getErrorCount());
+    }
+
+    private void addInvalidFixedAssetsToBalanceSheet() {
+        FixedAssets fixedAssets = new FixedAssets();
+        fixedAssets.setTangible(5L);
+        fixedAssets.setTotalFixedAssets(10L);
+
+        balanceSheet.setFixedAssets(fixedAssets);
```

Comment:
Could we have a `BeforeEach` set up method to reinstantiate the global `balanceSheet` object? This would avoid the risk of cross-contaminating test data.

Reply:
resolved in f8485a0